### PR TITLE
WARN: Add FutureWarning for `DataFrame.to_latex`

### PR DESCRIPTION
**Pull request by attack68**
Instead of #41648, which performs a refactor and adds the warning in the same PR, I am proposing adding the warning only for 1.4.0 and then in 2.0 there will be scope to make breaking changes and refactor the arguments signature.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3278,6 +3279,15 @@ def to_latex(
...
         \end{{tabular}}
         """
+        msg = (
+            "In future versions `DataFrame.to_latex` is expected to utilise the base "
+            "implementation of `Styler.to_latex` for formatting and rendering. "
+            "The arguments signature may therefore change. It is recommended instead to"
+            "use `DataFrame.style.to_latex` which also contains additional "
+            "functionality."
+        )
+        warnings.warn(msg, FutureWarning, stacklevel=2)
```
Comment (jreback) [inline pandas/core/generic.py:3289]: use `find_stack_level` here

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
looks fine can you add a whatsnew note in the deprecation section to message this. ping on green.

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -465,6 +465,7 @@ Other Deprecations
 - Deprecated the 'errors' keyword argument in :meth:`Series.where`, :meth:`DataFrame.where`, :meth:`Series.mask`, and meth:`DataFrame.mask`; in a future version the argument will be removed (:issue:`44294`)
 - Deprecated :meth:`PeriodIndex.astype` to ``datetime64[ns]`` or ``DatetimeTZDtype``, use ``obj.to_timestamp(how).tz_localize(dtype.tz)`` instead (:issue:`44398`)
 - Deprecated :meth:`DateOffset.apply`, use ``offset + other`` instead (:issue:`44522`)
+- A deprecation warning is now shown for :meth:`DataFrame.to_html` indicating the arguments signature may change and emulate more the arguments to :meth:`.Styler.to_html` in future versions (:issue:`44411`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:468]: hmm this should be for to_latex :->

**Comment by mroeschke** [comment]
@attack68 do you have bandwidth/appetite to align DataFrame.to_latex and Styler.to_latex since 2.0 will be the next version?

**Comment by attack68** [comment]
see https://github.com/pandas-dev/pandas/pull/47970#issuecomment-1320579957

**Pull request merged**